samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,274 @@
+            var authResultData = isTokenPresent ? "SSO success" : isStatePresent ? "OAuth success" : "SSO/OAuth failed";
+            
+            if (isBasicRefresh)
+            {
+                authResultData = "Refresh done";
+            }
+            
+            var payloadData = new
+            {
+                authResult = authResultData,
+            };
+
+            var cardJsonstring = template.Expand(payloadData);
+
+            var adaptiveCardResponse = new AdaptiveCardInvokeResponse()
+            {
+                StatusCode = 200,
+                Type = AdaptiveCard.ContentType,
+                Value = JsonConvert.DeserializeObject(cardJsonstring)
+            };
+
+            return CreateInvokeResponse(adaptiveCardResponse);
+        }
+
+        /// <summary>
+        /// when token is absent in the invoke. We can initiate SSO in response to the invoke
+        /// </summary>
+        /// <param name="turnContext">The context for the current turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task<InvokeResponse> initiateSSOAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var signInLink = await GetSignInLinkAsync(turnContext, cancellationToken).ConfigureAwait(false);
+            var oAuthCard = new OAuthCard
+            {
+                Text = "Signin Text",
+                ConnectionName = _connectionName,
+                TokenExchangeResource = new TokenExchangeResource
+                {
+                    Id = Guid.NewGuid().ToString()
+                },
+                Buttons = new List<CardAction>
+                    {
+                        new CardAction
+                        {
+                            Type = ActionTypes.Signin,
+                            Value = signInLink,
+                            Title = "Please sign in",
+                        },
+                    }
+            };
+
+            var loginReqResponse = JObject.FromObject(new
+            {
+                statusCode = 401,
+                type = "application/vnd.microsoft.activity.loginRequest",
+                value = oAuthCard
+            });
+
+            return CreateInvokeResponse(loginReqResponse);
+        }
+
+        /// <summary>
+        /// Get Adaptive Card
+        /// </summary>
+        /// <param name="filepath">json path</param>
+        /// <param name="signInLink">Get sign in link</param>
+        /// <param name="name">createdBy</param>
+        /// <param name="userMRI">createdById</param>
+        /// <returns></returns>
+        private Attachment GetAdaptiveCardWithAuthOptions(string[] filepath, string signInLink, string name = null, string userMRI = null)
```

Comment:
since this method returns both types of adaptive card, one with buttons to select card and with adaptive card with sso.

We should just rename it to GetAdaptiveCardFromFileName()

Reply:
Fixed